File: NancyDemo.VB/Security/AuthenticatedUser.vb
Imports Nancy.Security

Namespace Security

    Public Class AuthenticatedUser
        Implements IUserIdentity

        Public Property UserName() As String Implements IUserIdentity.UserName
        Public Property Claims() As IEnumerable(Of String) Implements IUserIdentity.Claims
        Public Property RealName As String
    End Class
End Namespace

File: NancyDemo.Csharp/Security/AuthenticatedUser.cs
using System;
using System.Collections.Generic;
using Nancy.Security;

namespace NancyDemo.Csharp.Security
{
    public class AuthenticatedUser : IUserIdentity
    {
        public String UserName { get; set; }
        public IEnumerable<String> Claims { get; set; } 
        public String RealName { get; set; }
    }
}